app/controllers/matches_controller.rb

```diff
@@ -54,6 +55,15 @@ def user_params
     params.require(:user).permit(:firstname, :lastname, :statement, :toc)
   end
 
+  def verify_redirect_to_other_match
+    if !@match.confirmable?
+      other = @match.find_other_available_match_for_user
+      if other
+        redirect_to Rails.application.routes.url_helpers.match_url(match_confirmation_token: other.match_confirmation_token, source: "redirect")
+      end
+    end
```

Comment:
If you like it, feel free to use guard clauses instead. I personally find them more readable but that's debatable 🙂 

```suggestion
    return if @match.confirmable?
    return unless (other = @match.find_other_available_match_for_user)
    
    redirect_to Rails.application.routes.url_helpers.match_url(match_confirmation_token: other.match_confirmation_token, source: "redirect")
```